codec-http2/src/main/java/io/netty5/handler/codec/http2/AbstractHttp2StreamChannel.java

```diff
@@ -933,4 +1009,17 @@ protected Future<Void> write0(ChannelHandlerContext ctx, Object msg) {
     protected abstract boolean isParentReadInProgress();
     protected abstract void addChannelToReadCompletePendingQueue();
     protected abstract ChannelHandlerContext parentContext();
+
+    /**
+     * Creates a new {@link ChannelOutputShutdownException} which has the origin of the given {@link Class} and method.
+     */
+    private static ChannelOutputShutdownException newOutputShutdownException(Class<?> clazz, String method) {
+        return ThrowableUtil.unknownStackTrace(new ChannelOutputShutdownException() {
+            @Override
+            public Throwable fillInStackTrace() {
+                // Suppress a warning since this method doesn't need synchronization
+                return this; // lgtm [java/non-sync-override]
+            }
+        }, clazz, method);
```

Comment:
There's a package-private duplicate of this code in `ChannelOutputShutdownException`. I wonder if we can collapse the two.

Reply:
lets just keep it for now as I didn't want to expose this as public